sdk/storage/azure-storage-file-share/azure/storage/fileshare/aio/_directory_client_async.py

```diff
@@ -94,11 +95,13 @@ def __init__(
             share_name: str,
             directory_path: str,
             snapshot: Optional[Union[str, Dict[str, Any]]] = None,
-            credential: Optional[Union[str, Dict[str, str], "AzureNamedKeyCredential", "AzureSasCredential", "TokenCredential"]] = None,  # pylint: disable=line-too-long
+            credential: Optional[Union[str, Dict[str, str], "AzureNamedKeyCredential", "AzureSasCredential", AsyncTokenCredential]] = None,  # pylint: disable=line-too-long
             *,
             token_intent: Optional[Literal['backup']] = None,
             **kwargs: Any
         ) -> None:
+        if isinstance(credential, AsyncTokenCredential) and token_intent is None:
+            raise ValueError("'token_intent' keyword must not be None if 'credential' is an AsyncTokenCredential.")
```

Comment:
Is this what we are envisoning? Open to feedback on placement of the check + the contents of the `ValueError`. Currently we cannot test this because of how we spoof OAuth in our testcases will not fall into this `isinstance`. See main description for more detail 😄

Reply:
Let's go with something like this; changing the check a little to handle empty string and slightly different wording:

```py
if isinstance(credential, AsyncTokenCredential) and not token_intent:
    raise ValueError("'token_intent' keyword is required when 'credential' is an AsyncTokenCredential.")
```